### ISSUE TITLE
Sequence.missing() not checking Item.exists()

### ISSUE THREAD

**Issue by herronelou** [rsgalloway/pyseq#44]
Hi there.
I've been using pyseq for a few tools over the last few months, but only today I realized Sequence.missing() is not actually checking that the items exist on disk, but just check the sequence for what has been passed as arguments when building the sequence.
I am using pyseq.uncompress() to generate sequences from arguments given by users (/path/to/sequence%04d.ext 0-10).
I was hoping to implement a check that would check this image sequence for missing frames, but sadly missing() returns [] even if I use fictional frame numbers.

Right now I'm using a small function I wrote which does:
```
missing = []
for i in seq:
    if not i.exists:
        missing.append(i)
for i in missing:
    seq.remove(i) 
```
I was wondering if there was maybe a way to have that in the api at all. (Sequence.prune() or even Sequence.missing(check_disk=True) or similar)

**Comment by rsgalloway** [comment]
Right, by default creating Sequences will not stat the filesystem, mostly for performance reasons. The lss tool works because it groups the sequences using only existing files from a listdir.

And uncompress is basically the reverse, it takes the string output of an lss, for example, and returns a Sequence object, but it still doesn't stat the filesystem. So missing() just returns the frames that were parsed from the string format. As you know, you can even use non-existent file sequences and it'll all still work.

There is are stat and exists properties on the Item class, so it should be easy to implement something that stats the filesystem on all the sequence members, like what you've done already. I'm not sure if uncompress is the place for that or maybe a new function would be better, or a new method on Sequence, or an argument to missing() as you suggest. I'm kind of partial to the last option, but maybe with a "stat" argument instead of "check_disk".

**Comment by nebukadhezer** [comment]
I have been using a function to glob for the specific image sequences since I have been using pyseq.
https://github.com/nebukadhezer/pyseq/blob/master/pyseq.py#L1478
egg in nuke to check for new frames or missing ones during a publish... But it presumes somethings about the padding and where it is placed in the string... so it is not as generic as uncompress but uncompress for example with a * wildcard could be a way to indicate that one has to glob...
uncompress("/path/to/sequence*.ext")

Is this the same usecase ?

**Comment by herronelou** [comment]
Hi Ryan,
I pretty much agree with all your points, but do feel that missing() is not really useful if it doesn't stat the filesystem, as if I have created the sequence via uncompress, I already know the missing frames since I have to pass them as an argument. This currently limits the use of missing() to sequences created with the get_sequences function.

Johannes,
My use case is slightly different but not very far. I'm currently writing a standalone tool to do batch image conversions, and wanted to to a quick check to verify the image sequence has no missing frames. 
I took a look at your code, and I might take a page or two from your book as I probably will have to support stereo as well soon. I do notice though that you have ["gouche","droit"] as a pair, but the proper french would be ["gauche","droit"], and we tend to use droite more than droit, but I guess if we're talking about an eye we might use 'droit'.

**Comment by rsgalloway** [comment]
I'm going to close this with the 0.9.0 RC, since it seems like the functionality is basically there, using one of the sequences in tests/files:

```bash
$ lss tests/files/file01.*.j2k
   3 file01.%03d.j2k [1-2, 4]
```

getting missing frames:

```python
>>> from pyseq import get_sequences, util
>>> seq = get_sequences("tests/files/file01.*.j2k")[0]  # get sequences from disk
>>> seq
<pyseq.Sequence "file01.1-4.j2k">
>>> seq.missing()
[3]
>>> seq = util.resolve_sequence("tests/files/file01.%03d.j2k")  # resolve sequence from sequence string
>>> seq
<pyseq.Sequence "file01.1-4.j2k">
>>> seq.missing()
[3]
```
